### DISCUSSION TITLE
Manual validation of field without showing error ?

### DISCUSSION THREAD

**Discussion by gytisgreitai** [react-hook-form/react-hook-form#2289] [Q&A]
Hi,

we are currently using the `onSubmit` mode which validates fields after user clicks submit, which is what we want from UX.

The problem is that we have this one field which we use to fetch data from backend and display the results. This must be done on change and only if that field is valid:

```javascript
 const value = watch('field')
 if (!errors['field']) {
    fetchFromBackend(value)
 }
```

Now obviously this does not work with `onSubmit` mode, cause the validation has not yet fired, and there is nothing in errors.
For now I see two ways to resolve this:

1. Use `onChange` validation. But this is a bit ugly from the UX, cause when user starts typing, error appears instantly
2. Validate using `yup` and then do something like `yup.reach(schema, 'field').validate(value)`

But maybe there is an option to invoke validation without using `yup` ? Something like...
```javascript
const { watch, validate } = useForm()

const value = watch('field')
const isValid = await validate('field')
if (isValid) {
 fetchFromBackend(value)
}
```

**Comment by bluebill1049** [comment]
trigger?

**Reply by gytisgreitai** [reply]
But `trigger` will display the error message to the user as soon as he starts typing, just the same as `onChange` mode ?

**Reply by bluebill1049** [reply]
that's up to you when to trigger it, onblur or combine with formState
